collectcontexts.py

```diff
@@ -32,40 +32,70 @@ def gather_chapter(api, book, chap):
         node = next_verse[0]
     return nodes
 
-def gather_book(api, book):
+def gather_book(api, book, lang):
     global VERSE_NODES
     result = dict()
     chap = 1
-    VERSE_NODES[book] = dict()
+    VERSE_NODES[lang][book] = dict()
     while True:
-        nodes = gather_chapter(api, book, chap)
+        nodes = gather_chapter(api, book, chap, lang)
         if nodes is None:
             return result
         result[chap] = nodes
         chap += 1
 
-def dump_book(api, book):
-    nodesets = gather_book(api, book)
+def dump_book(api, book, lang, use_features):
+    nodesets = gather_book(api, book, lang)
     for chap, nodes in nodesets.items():
         context = gather_context(
                 api,
-                {'features': FEATURES, 'locality': 'udnp'},
+                {'features': use_features, 'locality': 'udnp'},
                 (nodes,))
-        fname = book + '_' + str(chap) + '.pkl'
+        fname = lang + '_' + book + '_' + str(chap) + '.pkl'
         with open(os.path.join(DATADIR, fname), 'wb') as f:
             pickle.dump(context, f)
 
-def gather(locations, modules):
+def gather(locations, modules, lang):
+    global VERSE_NODES
     TF = Fabric(locations=locations, modules=modules, silent=True)
-    api = TF.load(FEATURES, silent=True)
+    if lang[0] == 'syriac':
+        use_features = SYR_FEATURES
+    elif lang[0] == 'greek':
+        use_features = GRK_FEATURES
+    else:
+        use_features = FEATURES
+    api = TF.load(use_features, silent=True)
+
+    VERSE_NODES[lang[0]] = {}
 
     for node in api.F.otype.s('book'):
         book = api.T.sectionFromNode(node)[0]
-        print(book)
-        dump_book(api, book)
+        dump_book(api, book, lang[0], use_features)
+
+    if lang[0] == 'hebrew':
+        with open(os.path.join(DATADIR, 'verse_nodes.pkl'), 'wb') as f:
+            pickle.dump(VERSE_NODES, f)
+    elif lang[0] == 'syriac':
+        with open(os.path.join(DATADIR, 'verse_nodes.pkl'), 'rb') as f:
+            HEB_VERSE_NODES = pickle.load(f)
+            FIN_VERSE_NODES = {
+                "hebrew": HEB_VERSE_NODES['hebrew'],
+                "syriac": VERSE_NODES['syriac']
+            }
 
-    with open(os.path.join(DATADIR, 'verse_nodes.pkl'), 'wb') as f:
-        pickle.dump(VERSE_NODES, f)
+            with open(os.path.join(DATADIR, 'verse_nodes.pkl'), 'wb') as f:
+                pickle.dump(FIN_VERSE_NODES, f)
+    elif lang[0] == 'greek':
+        with open(os.path.join(DATADIR, 'verse_nodes.pkl'), 'rb') as f:
+            PREV_VERSE_NODES = pickle.load(f)
+            FIN_VERSE_NODES = {
+                "hebrew": PREV_VERSE_NODES['hebrew'],
+                "syriac": PREV_VERSE_NODES['syriac'],
+                "greek": VERSE_NODES['greek']
+            }
+
+            with open(os.path.join(DATADIR, 'verse_nodes.pkl'), 'wb') as f:
+                pickle.dump(FIN_VERSE_NODES, f)
```

Comment:
It's good to do this in multiple steps (perhaps a user wants to use the program only for one data set), but now the program assumes that the data file is generated in a specific order: Hebrew > Syriac > Greek. Can we not just use separate files for each data set?